aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -105,4 +107,14 @@ public void initialize(Map<String, String> properties) {
     this.awsClientFactory = AwsClientFactories.from(properties);
     this.s3 = awsClientFactory::s3;
   }
+
+  @Override
+  public void close() {
+    // handles concurrent calls to close()
+    if (isResourceClosed.compareAndSet(false, true)) {
+      if (client != null) {
+        client.close();
+      }
```

Comment:
If this is checking for null (plus the compareAndSet), should it set it to null as well or is the AtomicBoolean sufficient?

Reply:
I think this is a good idea, but minor.